marvin.py

```diff
@@ -31,37 +31,42 @@ def get_page_title_from_url(page_url: str):
 # update. Error handlers also receive the raised TelegramError object in error.
 def start(bot, update):
     """Send a message when the command /start is issued."""
-    update.message.reply_text('Hi!')
-
-
-def help(bot, update):
-    """Send a message when the command /help is issued."""
-    update.message.reply_text('Help!')
+    update.message.reply_text('In un gruppo, rispondi ad un link con il comando /postalink')
 
 
-def postalink(bot, update):
+def postalink(subreddit, bot, update):
     # print("Reply from:" + str(update.message.reply_to_message.text))
-    message_entities_dict = update.message.reply_to_message.parse_entities([MessageEntity.URL])
-    print("Size of entities dict:" + str(len(message_entities_dict)))
-    # print("Author of the post: " + update.message.from_user.username)
-    print("Author of the post: " + update.message.from_user.name)
-    if len(message_entities_dict) == 1:
-        link_to_post = str(update.message.reply_to_message.parse_entity(next(iter(message_entities_dict))))
-        print("Link to post:" + link_to_post)
-        link_page_title = get_page_title_from_url(link_to_post)
-        print("Website title:" + link_page_title)
-        # submit to reddit:
-        global subreddit
-        # Create the post title
-        title = "[Post from telegram by:" + update.message.from_user.name + "]" + link_page_title
-        submission = subreddit.submit(title, url=link_to_post)
-        print("Link to created post:" + str(submission.shortlink))
-        update.message.reply_text("Post creato:" + str(submission.shortlink))
-    else:
-        update.message.reply_text("Non posso postare quel contenuto...")
-
-
-def error(bot, update, error):
+    if not update.message.reply_to_message:
+        update.message.reply_text("Per usare questo comando devi rispondere ad un messaggio")
+        return
+    message = update.message.reply_to_message
+    logger.info("Autore del messaggio: %s", message.from_user.name)
+
+    urls_entities = message.parse_entities([MessageEntity.URL])
+    print(urls_entities, len(urls_entities))
+    if not urls_entities:
+        update.message.reply_text("Il messaggio originale deve contenere una URL")
+        return
+    if len(urls_entities) > 1:
+        update.message.reply_text("Il messaggio originale deve contenere una **sola** URL")
+        return
+
+    link_to_post = next(iter(urls_entities.values()))
+    logger.debug("Link in message: %s", link_to_post)
+    link_parsed = urllib.parse.urlparse(link_to_post)
+    if link_parsed.scheme not in ['http', 'https']:
+        update.message.reply_text("Il messaggio originale deve contenere un link HTTP(S)")
+        return
+    link_page_title = get_page_title_from_url(link_to_post)
+    logger.debug("Link title from web: %s", link_page_title)
+    # Submit to reddit:
+    title = link_page_title + " [From telegram" + update.message.from_user.name + "]"
```

Comment:
Che fine ha fatto il commento che avevo messo qua?
E in ogni caso va prima la stringa con le parentesi quadre e l'utente e poi il titolo

Reply:
Il commento è nascosto da github perché ho fatto un secondo commit quindi viene considerato "outdate".

Io non metterei il "tag" [telegram]  prima del titolo, però sentiamo anche @fen0x che ne dice.

Reply:
Aiuto, non so cosa devo fare, questa cosa mi sta sfuggendo di mano.

In ogni caso, @timendum, ti ho mandato la richiesta per diventare contributore del progetto. 
Se volete possiamo usare il canale telegram di ItalyInformatica per parlarne: https://t.me/ItalyInformatica